### PR TITLE
Bump io.micrometer:micrometer-tracing-bom from 1.1.6 to 1.2.0

### PR DESCRIPTION
**Pull request by conghuhu**
## What is the purpose of the change

fix & close #13393 

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by songxiaosheng — APPROVED** [review]
LGTM

**Review by songxiaosheng — COMMENTED** [review]
Are you considering compatibility between two versions through the existence of classes?

**Comment by conghuhu** [comment]
> Are you considering compatibility between two versions through the existence of classes?

@songxiaosheng Thank you for your suggestion. The code has been updated

File: dubbo-spring-boot/dubbo-spring-boot-starters/observability/autoconfigure/src/main/java/org/apache/dubbo/spring/boot/observability/autoconfigure/otel/OpenTelemetryAutoConfiguration.java [hunk 1]
```diff
@@ -93,13 +95,31 @@ io.opentelemetry.sdk.trace.SdkTracerProvider otelSdkTracerProvider(
                 .getApplication()
                 .map(ApplicationConfig::getName)
                 .orElse(DEFAULT_APPLICATION_NAME);
+        // Due to https://github.com/micrometer-metrics/tracing/issues/343
+        boolean isLowVersion = false;
+        AttributeKey<String> attributeKey = null;
+        String SERVICE_NAME = "SERVICE_NAME";
+        try {
+            attributeKey = (AttributeKey<String>) ClassUtils.forName("io.opentelemetry.semconv.ResourceAttributes")
```
Comment (songxiaosheng) [inline dubbo-spring-boot/dubbo-spring-boot-starters/observability/autoconfigure/src/main/java/org/apache/dubbo/spring/boot/observability/autoconfigure/otel/OpenTelemetryAutoConfiguration.java:103]: please use `ClassUtils.isPresent(className, classLoader)` replace ClassUtils.forName

File: dubbo-spring-boot/dubbo-spring-boot-starters/observability/autoconfigure/src/main/java/org/apache/dubbo/spring/boot/observability/autoconfigure/otel/OpenTelemetryAutoConfiguration.java [hunk 2]
```diff
@@ -93,13 +95,31 @@ io.opentelemetry.sdk.trace.SdkTracerProvider otelSdkTracerProvider(
...
+                    .getDeclaredField(SERVICE_NAME)
+                    .get(null);
+        } catch (Throwable t) {
+            isLowVersion = true;
+        }
+        if (isLowVersion) {
+            try {
+                attributeKey = (AttributeKey<String>)
+                        ClassUtils.forName("io.opentelemetry.semconv.resource.attributes.ResourceAttributes")
```
Comment (songxiaosheng) [inline dubbo-spring-boot/dubbo-spring-boot-starters/observability/autoconfigure/src/main/java/org/apache/dubbo/spring/boot/observability/autoconfigure/otel/OpenTelemetryAutoConfiguration.java:112]: please use ClassUtils.isPresent(className, classLoader) replace ClassUtils.forName

**Review by songxiaosheng — CHANGES_REQUESTED** [review]

File: dubbo-spring-boot/dubbo-spring-boot-starters/observability/autoconfigure/src/main/java/org/apache/dubbo/spring/boot/observability/autoconfigure/otel/OpenTelemetryAutoConfiguration.java [hunk 3]
```diff
@@ -93,13 +95,29 @@ io.opentelemetry.sdk.trace.SdkTracerProvider otelSdkTracerProvider(
...
+        } catch (Throwable ignored) {
+        }
         io.opentelemetry.sdk.trace.SdkTracerProviderBuilder builder =
                 io.opentelemetry.sdk.trace.SdkTracerProvider.builder()
                         .setSampler(sampler)
                         .setResource(io.opentelemetry.sdk.resources.Resource.create(
-                                io.opentelemetry.api.common.Attributes.of(
-                                        io.opentelemetry.semconv.resource.attributes.ResourceAttributes.SERVICE_NAME,
-                                        applicationName)));
+                                io.opentelemetry.api.common.Attributes.of(attributeKey, applicationName)));
```
Comment (AlbumenJ) [inline dubbo-spring-boot/dubbo-spring-boot-starters/observability/autoconfigure/src/main/java/org/apache/dubbo/spring/boot/observability/autoconfigure/otel/OpenTelemetryAutoConfiguration.java:120]: `attributeKey` is nullable here?

Comment (conghuhu) [reply dubbo-spring-boot/dubbo-spring-boot-starters/observability/autoconfigure/src/main/java/org/apache/dubbo/spring/boot/observability/autoconfigure/otel/OpenTelemetryAutoConfiguration.java]: Assigned a default value

**Review by songxiaosheng — APPROVED** [review]
LGTM

**Pull request merged**